app/assets/javascripts/cart.js

```diff
@@ -0,0 +1,171 @@
+function shoppingCart(cartName) {
+    this.cartName = cartName;
+    this.items = [];
+    this.clearCart = false;
+
+    this.loadItems();
+    var self = this;
+    $(window).unload(function () {
+        if (self.clearCart) {
+            self.clearItems();
+        }
+        self.saveItems();
+        self.clearCart = false;
+    });
+}
+
+shoppingCart.prototype.loadItems = function () {
+    var items = localStorage !== null ? localStorage[this.cartName + "_items"] : null;
+    if (items !== null && JSON !== null) {
+        try {
+            items = JSON.parse(items);
+            for (var i = 0; i < items.length; i++) {
+                var item = items[i];
+                if (item.code !== null && item.name !== null && item.price !==
+                    null && item.quantity !== null &&
+                        item.image !== null) {
+                    item = new CartItem(
```

Comment:
Yeah. It just wasn't important for this feature. I guess other features
depend on it.

On Mon, Feb 1, 2016 at 3:54 PM, Emmanuel Chigbo <notifications@github.com>
wrote:

> In app/assets/javascripts/cart.js
> <https://github.com/andela/getmyshop/pull/40#discussion_r51426779>:
>
> > +        self.saveItems();
> > +        self.clearCart = false;
> > +    });
> > +}
> > +
> > +shoppingCart.prototype.loadItems = function () {
> > +    var items = localStorage !== null ? localStorage[this.cartName + "_items"] : null;
> > +    if (items !== null && JSON !== null) {
> > +        try {
> > +            items = JSON.parse(items);
> > +            for (var i = 0; i < items.length; i++) {
> > +                var item = items[i];
> > +                if (item.code !== null && item.name !== null && item.price !==
> > +                    null && item.quantity !== null &&
> > +                        item.image !== null) {
> > +                    item = new CartItem(
>
> I think we need to have the id of the item, some we can easily recognize
> the item at the backend and get more information about the item
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/andela/getmyshop/pull/40/files#r51426779>.
>



-- 
*Best Regards,*

*Yusuf Daniju /** Andela.com
<http://andela.com/?utm_source=andela&utm_medium=email&utm_campaign=signature> *
*Software Developer / **+234.807.266.7248 <+234.807.266.7248>** / @Andela
<http://bit.ly/andelatwitter>*

*"The startup that's harder to get into than Harvard" - *CNN
<http://snip.ly/HfId>
See what The Wall Street Journal <http://snip.ly/u9AI> has to say.